`JobPropertyStepTest` had been ignored for one reason but is now failing for another reason

### Jenkins and plugins versions report

Pipeline: Multibranch 696.v52535c46f4c9

### What Operating System are you using (both controller, and any agents involved in the problem)?

Ubuntu 20.04.3 LTS x86_64

### Reproduction steps

- Remove any exclusions for `JobPropertyStepTest`, if any remain in the source tree.
- Run: `PLUGINS=workflow-multibranch TEST=org.jenkinsci.plugins.workflow.multibranch.JobPropertyStepTest#configRoundTripParameters bash local-test.sh`

### Expected Results

The test passes.

### Actual Results

This regression went noticed because these test results were being ignored for a different reason (jenkinsci/workflow-multibranch-plugin#128). That PR was merged, but the test results were never un-ignored, so this new regression slipped by unnoticed.

This test now fails with:

```java
java.lang.AssertionError: expected:<1> but was:<3>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.jenkinsci.plugins.workflow.multibranch.JobPropertyStepTest.configRoundTripParameters(JobPropertyStepTest.java:116)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:601)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

### Anything else?

_No response_

--- a/bom-weekly/pom.xml
+++ b/bom-weekly/pom.xml
@@ -18,7 +18,7 @@
         <workflow-api-plugin.version>1108.v57edf648f5d4</workflow-api-plugin.version>
         <workflow-cps-plugin.version>2648.va9433432b33c</workflow-cps-plugin.version>
         <workflow-job-plugin.version>1145.v7f2433caa07f</workflow-job-plugin.version>
-        <workflow-multibranch-plugin.version>696.v52535c46f4c9</workflow-multibranch-plugin.version>
+        <workflow-multibranch-plugin.version>704.v8f039a_e2e8cf</workflow-multibranch-plugin.version>
         <workflow-step-api-plugin.version>622.vb_8e7c15b_c95a_</workflow-step-api-plugin.version>
         <workflow-support-plugin.version>804.vba10a18a1476</workflow-support-plugin.version>
         <plugin-util-api.version>2.12.0</plugin-util-api.version>

--- a/excludes.txt
+++ b/excludes.txt
@@ -19,8 +19,5 @@ org.jenkinsci.plugins.pipeline.modeldefinition.ToolsTest
 # TODO seemingly legitimate but undiagnosed PCT failures in interruptedWhileStartingMaxSurvivability and interruptedWhileStartingPerformanceOptimized
 org.jenkinsci.plugins.workflow.job.WorkflowRunRestartTest
 
-# TODO https://github.com/jenkinsci/bom/issues/811
-org.jenkinsci.plugins.workflow.multibranch.JobPropertyStepTest
-
 # TODO https://github.com/jenkinsci/ansicolor-plugin/pull/236
 hudson.plugins.ansicolor.action.ShortlogActionCreatorIntegrationTest

